### PR TITLE
[BEAM-3183] Add runner.run(transform) to Python SDK.

### PR DESCRIPTION
**Pull request by robertwb**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by robertwb** [comment]
R: @bkungfoo

File: sdks/python/apache_beam/runners/runner.py [hunk 1]
```diff
@@ -116,8 +116,21 @@ class PipelineRunner(object):
...
-    """Execute the entire pipeline or the sub-DAG reachable from a node."""
+  def run(self, transform, options=None):
+    """Run the given transform with this runner.
+    """
+    # Imported here to avoid circular dependencies.
+    # pylint: disable=wrong-import-order, wrong-import-position
+    from apache_beam.pipeline import Pipeline
+    p = Pipeline(runner=self, options=options)
+    p | transform
+    return p.run()
```
Comment (bkungfoo) [inline sdks/python/apache_beam/runners/runner.py]: In order to execute the transform eagerly, don't we have to add wait_until_finish?

result = p.run()
result.wait_until_finish()
return result

File: sdks/python/apache_beam/runners/runner.py [hunk 2]
```diff
@@ -116,8 +116,21 @@ class PipelineRunner(object):
   materialized values in order to reduce footprint.
   """
 
-  def run(self, pipeline):
-    """Execute the entire pipeline or the sub-DAG reachable from a node."""
+  def run(self, transform, options=None):
```
Comment (bkungfoo) [inline sdks/python/apache_beam/runners/runner.py]: Test cases? You can use runner_test.py in https://github.com/apache/beam/pull/4223/commits/9d87830c9ac311b8069a7e4a4f6470ba2e3223a6
and change runner.run() calls to runner.run_pipeline() if a pipeline is provided as input.

Comment (robertwb) [reply sdks/python/apache_beam/runners/runner.py]: This is tested at test_run_api. 

We don't want to accept runner.run(pipeline) as part of the public api.

File: sdks/python/apache_beam/runners/runner.py (hunk 1 above)
Comment (robertwb) [reply sdks/python/apache_beam/runners/runner.py]: This returns the result to the caller, which can wait (or not) as desired.

**Comment by charlesccychen** [comment]
This seems like a backwards-incompatible change that will break most existing pipelines.  What is the benefit of this over making this an extra argument to run, so that the existing pipeline construction strategy can be used, while allowing this new way to compose pipelines?

**Comment by robertwb** [comment]
Yes, this is only a change to the runner class. Users shouldn't be calling this directly.

**Pull request merged**